kayak_core/src/layout.rs

```diff
@@ -0,0 +1,68 @@
+use crate::layout_cache::Rect;
+use crate::Index;
+use morphorm::GeometryChanged;
+
+/// A layout event context sent to widgets
+#[derive(Debug, Default, Clone, Copy, PartialEq)]
+pub struct Layout {
+    /// width of the component
+    pub width: f32,
+    /// height of the component
+    pub height: f32,
+    /// x-coordinates of the component
+    pub x: f32,
+    /// y-coordinates of the component
+    pub y: f32,
+    /// z-coordinates of the component
+    pub z_index: f32,
```

Comment:
I think it would be better to name this just `z` since "index" implies that it's an integer.